xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/definitions.h

```diff
@@ -0,0 +1,1383 @@
+//----------------------------------------------------------------------------
+
+//============================================================================
+///
+/// \ingroup cpp_kodi_gui_CListItem_Defs
+/// @{
+/// @anchor CastProperties
+/// @brief Cast properties entry used on a list item
+typedef struct CastProperties
+{
+  /// Cast name, e.g. Michael C. Hall
+  std::string name;
+  /// Cast role, e.g. Dexter
+  std::string role;
+  /// Thumbnail URL, e.g. http://www.someurl.com/someimage.png
+  std::string thumbnail;
+  /// Order number
+  int order;
+} CastProperties;
+/// @}
+//----------------------------------------------------------------------------
+
+//============================================================================
+///
+/// \ingroup cpp_kodi_gui_CListItem_Defs
+/// @{
+/// @anchor FanartProperties
+/// @brief Fanart properties entry used on a list item
+typedef struct FanartProperties
+{
+  /// Image, e.g. http://www.someurl.com/someimage.png
+  std::string image;
+  /// Preview, e.g. http://www.someurl.com/somepreviewimage.png
+  std::string preview;
+  /// Colors (either comma separated Kodi hex values ("FFFFFFFF,DDDDDDDD") or
+  /// TVDB RGB Int Triplets ("|68,69,59|69,70,58|78,78,68|"))
+  std::string colors;
+} FanartProperties;
+/// @}
+//----------------------------------------------------------------------------
+
+//============================================================================
+///
+/// \ingroup cpp_kodi_gui_CControlEdit_Defs
+/// @{
+/// @anchor AddonGUIInputType
+/// @brief Text input types used on CControlEdit
+enum AddonGUIInputType
+{
+  /// Text inside edit control only readable
+  ADDON_INPUT_TYPE_READONLY = -1,
+  /// Normal text entries
+  ADDON_INPUT_TYPE_TEXT = 0,
+  /// To use on edit control only numeric numbers
+  ADDON_INPUT_TYPE_NUMBER,
+  /// To insert seconds
+  ADDON_INPUT_TYPE_SECONDS,
+  /// To insert time
+  ADDON_INPUT_TYPE_TIME,
+  /// To insert a date
+  ADDON_INPUT_TYPE_DATE,
+  /// Used for write in IP addresses
+  ADDON_INPUT_TYPE_IPADDRESS,
+  /// Text field used as password entry field with not visible text
+  ADDON_INPUT_TYPE_PASSWORD,
+  /// Text field used as password entry field with not visible text but
+  /// returned as MD5 value
+  ADDON_INPUT_TYPE_PASSWORD_MD5,
+  /// Use text field for search purpose
+  ADDON_INPUT_TYPE_SEARCH,
+  /// Text field as filter
+  ADDON_INPUT_TYPE_FILTER,
+  ///
+  ADDON_INPUT_TYPE_PASSWORD_NUMBER_VERIFY_NEW
+};
+/// @}
+//----------------------------------------------------------------------------
+
+//============================================================================
+///
+/// \ingroup kodi_addon_key_action_ids cpp_kodi_gui_CWindow_Defs
+/// @{
+/// @brief Actions that we have defined.
+///
+enum AddonGUIActionType
+{
+  ///
+  ADDON_ACTION_NONE                    = 0,
+
+  /// @brief Action string: <b><c>Left</c></b>
+  ///
+  /// Move left off a control.
+  ///
+  ADDON_ACTION_MOVE_LEFT               = 1,
+
```

Comment:
Have someone a good idea about them for action id's?

To do with a separate place bring advantages but also disadvantages.

I remove them now and hope to find a good way to interact with them and kodi and hold also independent.

With the "enum" has made the doxygen output a bit nicer as with a "define". :-)